Note about event listeners

Konrud opened the pull request:
Note that adding HTML elements using `innerHTML` will result in the removal of the previously set event listeners.
Appreciate any addition / correction of the note.

Konrud committed:
Note about event listeners

Note that adding HTML elements using `innerHTML` will result in the removal of the previously set event listeners.

wbamberg commented:
@Konrud , I thought this was surprising, so I tried it out, and did reproduce this in Firefox but couldn't in Chrome. Do you know if this is specced behavior? Have you reproduced  it in Chrome?

Konrud commented:
I've created a [test](https://jsfiddle.net/Konrud/gxpn4om5/). 
 I've checked it in:
 * **Chrome** Version 94.0.4606.61 (Official Build) (64-bit)  [Windows 10]
 * **Firefox** Version 92.0 (64-bit) [Windows 10]
 * **Opera** Version:79.0.4143.50 [Windows 10]
 * **Edge** Version 94.0.992.31 (Official build) (64-bit) [Windows 10]
 
In all of them, I've been able to reproduce this very behavior.

As for the specced behavior, I'm not sure it's even in the spec (at least I couldn't find it). I believe it has to do with the way `innerHTML`  works. That is, by using `+=` we are actually taking previously created elements in the DOM, convert them into the string representation, append another data to the string, and add it to the DOM as a string that will be parsed by the UA. I think that's why all the event listeners are removed. We are working with the DOM elements represented as a string rather than Nodes.

wbamberg reviewed (changes requested):
Thanks for creating a fiddle for this! I can reproduce this on Chrome. Before, I was using the Chrome devtools, in which I could not, and still can't, reproduce.

I agree we should document this, but I don't think a note at the start is the right way. At the moment we have:

```
H3 Usage notes
    H4 Reading the HTML contents of an element
    H4 Replacing the contents of an element
    H4 Security considerations
```

I think we should add an H4:

```
H3 Usage notes
    H4 Reading the HTML contents of an element
    H4 Replacing the contents of an element
    H4 Appending to an element
    H4 Security considerations
```

...and in there have a line or two about this, and a simple example like the existing ones, and note this issue about event listeners. I don't think it needs the `> **Note:**` style, it can just be a paragraph.

Konrud committed:
Change note into code example

Note about appending new elements via `innerHTML`:

> **Note:** If a {{HTMLElement("div")}},
> {{HTMLElement("span")}}, or {{HTMLElement("noembed")}} node has a child text node that
> includes the characters `(&)`, `(<)`, or
> `(>)`, `innerHTML` returns these characters as the HTML
> entities `"&amp;"`, `"&lt;"` and `"&gt;"`
> respectively. Use {{domxref("Node.textContent")}} to get a raw copy of these text nodes'
> contents.


 has been removed in favour of a paragraph with an example.

Konrud commented:
@wbamberg, I've changed it according to your suggestion. Please let me know if it's sufficient or should be extended/changed.

Konrud committed:
Remove note

wbamberg commented on files/en-us/web/api/element/innerhtml/index.md:
```diff
@@ -112,6 +104,30 @@ user agent to follow these steps:
 3.  For all other elements, the element's contents are replaced with the nodes in the
     new `DocumentFragment`.
+    
+### Appending the contents to an element  
```
```suggestion
### Appending HTML to an element  
```

wbamberg committed:
Clarify heading

wbamberg reviewed (approved):
Looks great, thank you @Konrud !
